### ISSUE TITLE
Suggestion on RX/TX decoupling

### ISSUE THREAD

**Issue by svyatoslavnik** [ultraethernet/uet-ref-prov#56]
**Revised Version:**

The current implementation of the RX and TX processes is handled within `uet_cq_read`, and we understand the reasoning behind this. However, this approach limits the ability to test and explore some functionalities, and may also lead to issues such as buffer overflow or bloating.

We suggest breaking down the TX and RX processes into smaller units and executing them either concurrently or semi-concurrently. Here are the options we see to achieve this:

1.     Place `progress_rx` and `progress_tx` not only in `uet_cq_read`, but in all functions related to data flow (send, receive, etc.). This quick solution could be implemented relatively easily and would unblock certain functionalities.
2.     Implement separate threads for RX and TX processes, spawning them during initialization. These threads would allow the buffers to be processed independently of the libfabric API calls (such as reading CQ, posting send and receive).

What are your thoughts on this approach?

**Comment by insanum** [comment]
I agree that splitting them apart offers much more flexibility. I think our initial implementation simply followed what progress calls are available in libfabric.

**Comment by svyatoslavnik** [comment]
Yes, exactly! It provides the most important quality for any code – it works as expected. The key is to follow the simple rule: pull CQ using the provided API, and everything should function smoothly.
Let's further improve it : )
